src/components/filters/filter_icons_overlay/fitler_icons_overlay.ts

```diff
@@ -31,15 +31,37 @@ export class FilterIconsOverlay extends Component<Props, SpreadsheetChildEnv> {
 
     const rowDims = this.env.model.getters.getRowDimensionsInViewport(sheetId, position.row);
     const colDims = this.env.model.getters.getColDimensionsInViewport(sheetId, position.col);
+    const cell = this.env.model.getters.getCell({ sheetId, ...position });
+    const verticalFilterIconPosition = this.getIconVerticalPosition(rowDims, cell);
 
-    // TODO : change this offset when we support vertical cell align
-    const centeringOffset = Math.floor((rowDims.size - FILTER_ICON_EDGE_LENGTH) / 2);
     return {
       x: colDims.end - FILTER_ICON_EDGE_LENGTH + this.props.gridPosition.x - FILTER_ICON_MARGIN - 1, // -1 for cell border
-      y: rowDims.end - FILTER_ICON_EDGE_LENGTH + this.props.gridPosition.y - centeringOffset,
+      y: verticalFilterIconPosition + this.props.gridPosition.y,
     };
   }
 
+  // Calculates the vertical position of the filter icon based on the row dimensions and cell styles.
+  private getIconVerticalPosition(
+    rowDims: { start: number; end: number; size: number },
+    cell: any
+  ): number {
+    const centeringOffset = Math.floor((rowDims.size - FILTER_ICON_EDGE_LENGTH) / 2);
+    let verticalFilterIconPosition = rowDims.end - FILTER_ICON_EDGE_LENGTH - centeringOffset;
+
+    switch (cell?.style?.verticalAlign) {
+      case "bottom":
+        verticalFilterIconPosition = rowDims.end - FILTER_ICON_MARGIN - FILTER_ICON_EDGE_LENGTH;
```

Comment:
```suggestion
        return rowDims.end - FILTER_ICON_MARGIN - FILTER_ICON_EDGE_LENGTH;
```
no need fot an intermediate variable